edxval/tests/test_api.py

```diff
@@ -2030,6 +2038,54 @@ def test_delete_video_transcript(self, mock_logger):
             query_filter['language_code']
         )
 
+    def test_create_transcript_file(self):
+        """
+        Tests that transcript file is created correctly.
+        """
+        root_dir = path(self.temp_dir)
+        file_system = OSFS(root_dir)
+        file_system.makedir(STATIC_DIR, recreate=True)
+
+        # Create transcript file now.
+        api.create_trancript_file(
+            edx_video_id=constants.VIDEO_DICT_FISH['edx_video_id'],
+            language_code='en',
+            file_format=TranscriptFormat.SRT,
+            transcript_dir=STATIC_DIR,
+            resource_fs=file_system
+        )
+
+        # Verify transcript is file created.
+        self.assertTrue(u'super-soaker-en.srt' in file_system.listdir(STATIC_PATH))
+
+    @data(
+        ('invalid-video-id', 'invalid-language-code', None, False),
+        ('super-soaker', 'invalid-language-code', None, False),
+        ('super-soaker', 'en', None, True),
+        ('super-soaker', 'en', STATIC_DIR, False)
+    )
+    @unpack
+    def test_no_create_transcript_file(self, video_id, language_code, transcript_dir, is_resource_fs):
```

Comment:
If missing one of the `transcript_dir` or `is_resource_fs`, is an invalid scenario then, we should not make them optional in the first place and this test wont also be needed.